app/routes/synonyms.$synonymId.tsx

```diff
@@ -148,7 +207,7 @@ export default function () {
           <input type="hidden" name="deleteSynonyms" value={deleteSynonyms} />
           <input type="hidden" name="addSynonyms" value={addSynonyms} />
```

Comment:
Since these state variables are both sets, I think you can just emit an input for each of their members, for easier deserialization.

According to (for example) https://webmasters.stackexchange.com/questions/9775/is-it-valid-to-have-multiple-form-input-elements-with-the-same-name, you should get all of the values of identically-named inputs in the form data.

Reply:
I'm not sure I fully understand what you are asking for here. 
Do you mean doing something like this?
```
{addSynonyms.map((synonym)=>(<input key={synonym} type="hidden" name="addSynonyms" value={synonym} id={`add-${synonym}`}/>))}
{deleteSynonyms.map((synonym)=>(<input key={synonym} type="hidden" name="deleteSynonyms" value={synonym} id={`del-${synonym}`}/>))}
```
If so, that does not work, it just sends the first value and does not return an array of multiple values.

Reply:
Starting state:
<img width="278" alt="Screenshot 2024-10-10 at 11 21 40 AM" src="https://github.com/user-attachments/assets/db717d3c-58ae-4808-a66f-1d7765e66bc2">

Two removed, two added:
<img width="290" alt="Screenshot 2024-10-10 at 11 23 12 AM" src="https://github.com/user-attachments/assets/294759f0-6af7-49f9-ac7e-1c76ee2388e7">

The code where I am logging in the action:
<img width="663" alt="Screenshot 2024-10-10 at 11 23 36 AM" src="https://github.com/user-attachments/assets/3b678d53-dd86-4cd4-9c3d-6c4ad2a0954e">

The terminal output:
<img width="293" alt="Screenshot 2024-10-10 at 11 23 23 AM" src="https://github.com/user-attachments/assets/e0e2e96f-3963-4f07-831f-e65521d0c2cd">

Reply:
That might be a limitation of our own function, `getFormDataString`. I'm going to do a little experiment with my browser. Will report back in a moment...

Reply:
Take the following little HTML page, `test.html`:

```html
<!DOCTYPE html>
<form>
  <input name="foo" value="bar">
  <input name="foo" value="bat">
  <input type="submit">
</form>
```

Open it in a browser and click the submit button. The address bar updates to `file:///.../test.html?foo=bar&foo=bat`, showing that at least for URL-encoded form submission, both values are included.